CMakeLists.txt

```diff
@@ -258,8 +257,18 @@ endif()
 
 option(WITH_BLUESTORE "Bluestore OSD backend" ON)
 if(WITH_BLUESTORE)
-  find_package(aio)
-  set(HAVE_LIBAIO ${AIO_FOUND})
+  if(LINUX)
+    find_package(aio)
+    set(HAVE_LIBAIO ${AIO_FOUND})
+    set(HAVE_POSIXAIO OFF)
+  elseif(FREEBSD)
+    # the AIO library is native in libc
```

Comment:
IMHO, it would be easier to digest if put this way:

> POSIX AIO is integrated into FreeBSD kernel, and exposed by libc.

FreeBSD has no such a libaio as GNU/Linux does. see also 
https://github.com/ceph/ceph/pull/18357#issuecomment-337930255